<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Compute deconvolution — compute.deconvolution • multideconv</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Compute deconvolution — compute.deconvolution"><meta name="description" content="The function calculates cell abundance based on cell type signatures using different methods and signatures. Methods available are Quantiseq, MCP, XCell, CibersortX, EpiDISH, DWLS and DeconRNASeq. Provided signatures included signatures based on bulk and methylation data (7 methods and 10 signature in total). Signatures are present in the src/signatures directory, user can add its own signatures by adding the .txt files in this same folder. Second generation methods to perform deconvolution based on single cell data are also available if scRNAseq object is provided."><meta property="og:description" content="The function calculates cell abundance based on cell type signatures using different methods and signatures. Methods available are Quantiseq, MCP, XCell, CibersortX, EpiDISH, DWLS and DeconRNASeq. Provided signatures included signatures based on bulk and methylation data (7 methods and 10 signature in total). Signatures are present in the src/signatures directory, user can add its own signatures by adding the .txt files in this same folder. Second generation methods to perform deconvolution based on single cell data are also available if scRNAseq object is provided."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">multideconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/multideconv.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/VeraPancaldiLab/multideconv/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Compute deconvolution</h1>
      <small class="dont-index">Source: <a href="https://github.com/VeraPancaldiLab/multideconv/blob/main/R/cell_deconvolution.R" class="external-link"><code>R/cell_deconvolution.R</code></a></small>
      <div class="d-none name"><code>compute.deconvolution.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The function calculates cell abundance based on cell type signatures using different methods and signatures. Methods available are Quantiseq, MCP, XCell, CibersortX, EpiDISH, DWLS and DeconRNASeq. Provided signatures included signatures based on bulk and methylation data (7 methods and 10 signature in total). Signatures are present in the src/signatures directory, user can add its own signatures by adding the .txt files in this same folder. Second generation methods to perform deconvolution based on single cell data are also available if scRNAseq object is provided.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">compute.deconvolution</span><span class="op">(</span></span>
<span>  <span class="va">raw.counts</span>,</span>
<span>  methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Quantiseq"</span>, <span class="st">"CBSX"</span>, <span class="st">"Epidish"</span>, <span class="st">"DeconRNASeq"</span>, <span class="st">"DWLS"</span>, <span class="st">"MOMF"</span><span class="op">)</span>,</span>
<span>  signatures_exclude <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  normalized <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  doParallel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  workers <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  return <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  create_signature <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  credentials.mail <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  credentials.token <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sc_deconv <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  sc_matrix <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sc_metadata <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cell_label <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sample_label <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cell_markers <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  name_sc_signature <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  file_name <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-raw-counts">raw.counts<a class="anchor" aria-label="anchor" href="#arg-raw-counts"></a></dt>
<dd><p>A matrix with the raw counts (samples as columns and genes symbols as rows)</p></dd>


<dt id="arg-methods">methods<a class="anchor" aria-label="anchor" href="#arg-methods"></a></dt>
<dd><p>A character vector with the deconvolution methods to run. Default are "Quantiseq", "MCP", "xCell", "CBSX", "Epidish", "DeconRNASeq", "DWLS"</p></dd>


<dt id="arg-signatures-exclude">signatures_exclude<a class="anchor" aria-label="anchor" href="#arg-signatures-exclude"></a></dt>
<dd><p>A character vector with the signatures to exclude from the src/signatures folder.</p></dd>


<dt id="arg-normalized">normalized<a class="anchor" aria-label="anchor" href="#arg-normalized"></a></dt>
<dd><p>If raw.counts are not available, user can input its normalized counts. In that case this argument need to be set to False.</p></dd>


<dt id="arg-doparallel">doParallel<a class="anchor" aria-label="anchor" href="#arg-doparallel"></a></dt>
<dd><p>Whether to do or not parallelization. Only CBSX and DWLS methods will run in parallel.</p></dd>


<dt id="arg-workers">workers<a class="anchor" aria-label="anchor" href="#arg-workers"></a></dt>
<dd><p>Number of processes available to run on parallel. If no number is set, this will correspond to detectCores() - 1</p></dd>


<dt id="arg-return">return<a class="anchor" aria-label="anchor" href="#arg-return"></a></dt>
<dd><p>Whether to save or not the csv file with the deconvolution features</p></dd>


<dt id="arg-create-signature">create_signature<a class="anchor" aria-label="anchor" href="#arg-create-signature"></a></dt>
<dd><p>Whether to create or not the signatures using the methods MOMF, CBSX, DWLS and BSeq-SC. If TRUE, sc_matrix shuld be provide.</p></dd>


<dt id="arg-credentials-mail">credentials.mail<a class="anchor" aria-label="anchor" href="#arg-credentials-mail"></a></dt>
<dd><p>(Optional) Credential email for running CibersortX. If not provided, cibersortX method will not be run.</p></dd>


<dt id="arg-credentials-token">credentials.token<a class="anchor" aria-label="anchor" href="#arg-credentials-token"></a></dt>
<dd><p>(Optional) Credential token for running CibersortX. If not provided, cibersortX method will not be run.</p></dd>


<dt id="arg-sc-deconv">sc_deconv<a class="anchor" aria-label="anchor" href="#arg-sc-deconv"></a></dt>
<dd><p>Whether to run or not deconvolution methods based on single cell.</p></dd>


<dt id="arg-sc-matrix">sc_matrix<a class="anchor" aria-label="anchor" href="#arg-sc-matrix"></a></dt>
<dd><p>If sc_deconv = T, the matrix of counts across cells from the scRNAseq object is provided.</p></dd>


<dt id="arg-sc-metadata">sc_metadata<a class="anchor" aria-label="anchor" href="#arg-sc-metadata"></a></dt>
<dd><p>Dataframe with metadata from the single cell object. The matrix should include the columns cell_label and sample_label.</p></dd>


<dt id="arg-cell-label">cell_label<a class="anchor" aria-label="anchor" href="#arg-cell-label"></a></dt>
<dd><p>If sc_deconv = T, a character vector indicating the cell labels (same order as the count matrix)</p></dd>


<dt id="arg-sample-label">sample_label<a class="anchor" aria-label="anchor" href="#arg-sample-label"></a></dt>
<dd><p>If sc_deconv = T, a character vector indicating the cell samples IDs (same order as the count matrix)</p></dd>


<dt id="arg-cell-markers">cell_markers<a class="anchor" aria-label="anchor" href="#arg-cell-markers"></a></dt>
<dd><p>Named list with the genes markers names as Symbol per cell types to be used to create the signature using the BSeq-SC method. If NULL, the method will be ignored during the signature creation.</p></dd>


<dt id="arg-name-sc-signature">name_sc_signature<a class="anchor" aria-label="anchor" href="#arg-name-sc-signature"></a></dt>
<dd><p>If sc_deconv = T, the name you want to give to the signature generated</p></dd>


<dt id="arg-file-name">file_name<a class="anchor" aria-label="anchor" href="#arg-file-name"></a></dt>
<dd><p>File name for the csv files and plots saved in the Results/ directory</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A matrix of cell type deconvolution features across samples</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Sturm, G., Finotello, F., Petitprez, F., Zhang, J. D., Baumbach, J., Fridman, W. H., ..., List, M., Aneichyk, T. (2019). Comprehensive evaluation of transcriptome-based cell-type quantification methods for immuno-oncology.
Bioinformatics, 35(14), i436-i445. https://doi.org/10.1093/bioinformatics/btz363</p>
<p>Benchmarking second-generation methods for cell-type deconvolution of transcriptomic data. Dietrich, Alexander and Merotto, Lorenzo and Pelz, Konstantin and Eder, Bernhard and Zackl, Constantin and Reinisch, Katharina and
Edenhofer, Frank and Marini, Federico and Sturm, Gregor and List, Markus and Finotello, Francesca. (2024) https://doi.org/10.1101/2024.06.10.598226</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"raw_counts"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"cell_labels"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"sample_labels"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metacells_data"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metacells_metadata"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"pseudobulk"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">deconv</span> <span class="op">=</span> <span class="fu">compute.deconvolution</span><span class="op">(</span><span class="va">raw_counts</span>, normalized <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                               methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Epidish"</span>, <span class="st">"DeconRNASeq"</span><span class="op">)</span>, return <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Performing TPM normalization ................................................................................</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting input to matrix.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running deconvolution using the following methods...............................................................</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Epidish</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * DeconRNASeq</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in compute_methods_variable_signature(TPM_matrix, signatures = path_signatures,     algos = methods, exclude = signatures_exclude, cbsx.name = credentials.mail,     cbsx.token = credentials.token, doParallel, workers, sc_matrix):</span> object 'signature_dir' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Marcelo Hurtado.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer></div>





  </body></html>

